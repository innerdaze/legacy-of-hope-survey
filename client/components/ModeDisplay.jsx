import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Title from 'grommet/components/Title'
import modes from '../constants/OperationModes'
import modeNames from '../constants/OperationModeNames'

class ModeDisplay extends Component {
  constructor(props) {
    super(props)

    this.state = {
      mode: props.mode
    }
  }

  componentWillReceiveProps(newProps) {
    if (newProps.mode === this.state.mode) {
      return false
    } else {
      this.setState({
        mode: newProps.mode
      })
    }
  }

  render() {
    return (
      <Title>
        {modeNames[this.state.mode]}
      </Title>
    )
  }
}

ModeDisplay.propTypes = {
  mode: PropTypes.number.isRequired
}

ModeDisplay.defaultProps = {
  mode: modes.ORDERING
}

export default ModeDisplay
