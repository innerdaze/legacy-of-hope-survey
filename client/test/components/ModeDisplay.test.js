import React from 'react'
import ModeDisplay from '../../components/ModeDisplay'
import modeNames from '../../constants/OperationModeNames'

let wrapper, mode

describe('Testing on ModeDisplay...', () => {
  beforeEach(() => {
    mode = parseInt(Object.keys(modeNames)[0])
    wrapper = shallow(<ModeDisplay mode={mode} />)
  })
  describe('ModeDisplay layout', () => {
    test('Expect ModeDisplay to have a Title', () => {
      expect(wrapper.find('Title')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect Title to display Delivery if mode prop change to 20', () => {
      mode = 20
      wrapper = render(<ModeDisplay mode={mode} />)

      expect(wrapper.text()).toEqual('Delivery')
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on state', () => {
    test('Expect state to change new mode prop is injected', () => {
      wrapper = mount(<ModeDisplay mode={mode} />)
      wrapper.setState({
        mode: 10
      })
      wrapper.setProps({
        mode: 20
      })

      expect(wrapper.state('mode')).toEqual(20)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
