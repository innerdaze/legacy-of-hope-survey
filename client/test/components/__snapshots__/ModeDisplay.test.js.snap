// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on ModeDisplay... ModeDisplay layout Expect ModeDisplay to have a Title 1`] = `
<Title
  responsive={true}
  truncate={true}
>
  Ordering
</Title>
`;

exports[`Testing on ModeDisplay... Test on prop Expect Title to display Delivery if mode prop change to 20 1`] = `
<div
  class="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-none grommetux-title grommetux-title--responsive grommetux-title--truncate"
>
  <span>
    Delivery
  </span>
</div>
`;

exports[`Testing on ModeDisplay... Test on state Expect state to change new mode prop is injected 1`] = `
<ModeDisplay
  mode={20}
>
  <Title
    responsive={true}
    truncate={true}
  >
    <Box
      a11yTitle="Title"
      align="center"
      announce={false}
      className="grommetux-title grommetux-title--responsive grommetux-title--truncate"
      direction="row"
      focusable={true}
      pad="none"
      responsive={false}
      tag="div"
    >
      <div
        className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-none grommetux-title grommetux-title--responsive grommetux-title--truncate"
        style={Object {}}
      >
        <span>
          Delivery
        </span>
      </div>
    </Box>
  </Title>
</ModeDisplay>
`;
